### PR TITLE
Hipchat enterprise import

### LINKED ISSUE #2997: Import hipchat enterprise data
Your Rocket.Chat version: 0.27.0

When importing data from a hipchat server to Rocket chat, rocket chat requires the file to be compressed in a zip format but on hipchat 1.3.9 (2016.03.24.210917) the file is in a tar.gz.aes format which the rocket chat importer does not recognise.

I decrypt the file with openssl aes-256-cbc -d -in hipchat-2016-04-21_17-22-52.tar.gz.aes -out hipchat-2016-04-21_17-22-52.tar.gz -pass pass:password so it's left with only tar.gz this also fails as rocket chat only wants a zip file which hipchat does not give anymore.

Extracting the tar.gz and compressing the files to a zip file does not work.

Could this be looked into? I imagine that there are users who are using this feature so it would be great if it could be fixed.

### PR DESCRIPTION
**Pull request by graywolf336**
@RocketChat/core 

Closes #2997

This pull request will add support for importing of the Hipchat Enterprise data, as it is a different format and structure than the hipchat cloud.

### CODE REVIEW

**Comment by graywolf336** [comment]
@JamieH mind testing this when you get a chance? Thanks!

**Comment by engelgabriel** [comment]
@graywolf336 can we merge this and open new issues if we find bugs?

**Comment by graywolf336** [comment]
@engelgabriel Yes, however let me add another warning to it stating that it's in beta and they should report issues before you merge it.

**Comment by graywolf336** [comment]
@engelgabriel Once the build completes and everything looks fine, should be good to go

File: packages/rocketchat-importer/client/admin/adminImportPrepare.coffee [hunk 1]
```diff
@@ -23,13 +23,16 @@ Template.adminImportPrepare.helpers
...
+		return if not importer.key
+
 		e = event.originalEvent or event
 		files = e.target.files
 		if not files or files.length is 0
 			files = e.dataTransfer?.files or []
 
 		for blob in files
 			template.preparing.set true
+			console.log blob.type, importer.fileTypeRegex
```
Comment (rodrigok) [inline packages/rocketchat-importer/client/admin/adminImportPrepare.coffee]: Can you remove this log?

**Comment by graywolf336** [comment]
@engelgabriel if nothing else is blocking the merge in the code, then it's good to go and people can start using/testing it.

**Pull request merged**